### ISSUE TITLE
`unnecessary_parenthesis`: False positive around switch expressions

### ISSUE THREAD

**Issue by hacker1024** [dart-lang/linter#4361]
**Describe the issue**
A switch expression can be written in place of a statement by using parentheses. This is triggering the `unnecessary_parenthesis` lint.

**To Reproduce**

```dart
void main() {
  (switch (0) {
    0 => "a",
    1 => "b",
    _ => "c",
  });
}
```

**Expected behavior**
The `unnecessary_parenthesis` should not be reported in this context.

**Additional context**

**Comment by srawlins** [comment]
_Great_ catch!

As for an actual use of a switch expression statement, I can imagine void/Never-typed results, like throwing this or that.

**Comment by srawlins** [comment]
Closed with https://github.com/dart-lang/sdk/commit/81bc23067e304fa1520fecbead8a94ea6f6a217c
